lapis/src/main/kotlin/org/genspectrum/lapis/controller/LapisController.kt

```diff
@@ -894,6 +899,255 @@ class LapisController(
             },
         )
 
+    @GetMapping(MOST_RECENT_COMMON_ANCESTOR_ROUTE, produces = [MediaType.APPLICATION_JSON_VALUE])
+    @LapisMostRecentCommonAncestorResponse
+    fun getMostRecentCommonAncestorAsJson(
+        @PrimitiveFieldFilters
+        @RequestParam
+        sequenceFilters: GetRequestSequenceFilters?,
+        @PhyloTreeField
+        @RequestParam
+        phyloTreeField: String,
+        @Parameter(
+            description = PRINT_NODES_NOT_IN_TREE_FIELD_DESCRIPTION,
+        )
+        @RequestParam(required = false, defaultValue = "false")
+        printNodesNotInTree: Boolean,
+        @NucleotideMutations
+        @RequestParam
+        nucleotideMutations: List<NucleotideMutation>?,
+        @AminoAcidMutations
+        @RequestParam
+        aminoAcidMutations: List<AminoAcidMutation>?,
+        @DataFormat
+        @RequestParam
+        dataFormat: String? = null,
+        @NucleotideInsertions
+        @RequestParam
+        nucleotideInsertions: List<NucleotideInsertion>?,
+        @AminoAcidInsertions
+        @RequestParam
+        aminoAcidInsertions: List<AminoAcidInsertion>?,
+        response: HttpServletResponse,
+    ) {
+        val request = PhyloTreeSequenceFiltersRequest(
+            sequenceFilters?.filterKeys { !SPECIAL_REQUEST_PROPERTIES.contains(it) }
+                ?: emptyMap(),
+            nucleotideMutations ?: emptyList(),
+            aminoAcidMutations ?: emptyList(),
+            nucleotideInsertions ?: emptyList(),
+            aminoAcidInsertions ?: emptyList(),
+            phyloTreeField,
+            printNodesNotInTree = printNodesNotInTree,
+        )
+
+        lapisResponseStreamer.streamData(
+            request = request,
+            getData = ::getMostRecentCommonAncestorCollection,
+            response = response,
+            responseFormat = ResponseFormat.Json,
+        )
+    }
+
+    @GetMapping(MOST_RECENT_COMMON_ANCESTOR_ROUTE, produces = [TEXT_CSV_VALUE])
+    @StringResponseOperation(
+        operationId = "getMostRecentCommonAncestorAsCsv",
+    )
+    fun getMostRecentCommonAncestorAsCsv(
```

Comment:
I mean to recall that we said that we don't need CSV for this action since it always only returns a single line? Isn't that true anymore? 

If it was only ever a single line, then it would also make more sense to return no array, but an object.

I.e.:
```
{
  "data": {
      "mrcaNode": "NODE_0000000",
      "missingNodeCount": 2,
      "missingFromTree": null
  }
  ,
  "info": {...}
}
```

instead of 
```
{
  "data": [
    {
      "mrcaNode": "NODE_0000000",
      "missingNodeCount": 2,
      "missingFromTree": null
    }
  ],
  "info": {...}
}
```

Reply:
This might be nicer but it would require a new LapisResponseStreamer which doesn't expect a list of records - so I would prefer to refactor this in a later PR as this is already getting really long. I'm also not sure if it might be better to have the same output structure for all requests - @chaoran-chen what do you think the user will expect?

Reply:
Both solutions are fine and I'm happy with keeping `data: []` since it is already implemented. Taking the first element is an acceptable burden and if because `data: []` is conform with the other endpoints, people might be able to reuse code more easily.

Reply:
Then let's create an issue and discuss in the next Tuesday meeting whether we want to do it?